chrisj117/CS-472-GROUP-2-PROJECT
# Frontend Auth Completed

**_WHAT DOES THIS PR DO?_**

  - Fixes registration bugs
  - Fixes login bugs
  - Adds error handling for login
  - Adds profile popup for logged in users
  - Protects login and registration routes if signed in
  - Creates and protects profile route
  - Adds link if forgot password on login page (but no route yet)
  - Adds loading phases for login, registration, logout
  - Adds auth check on review page
  - Fixes styling issues related to navbar

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

  - #43 
  - #99 
  - #43 
  - #101 
  - #131 

**_HOW DO I TEST OUT THIS PR?_**

  - Make sure the local server is running using `python3 manage.py runserver` in the `api` directory. Head over to the `web` directory and run `npm i` and `npm run dev` to run the frontend. You can test the registration by going to login, clicking "Create an account" or just go to `registration` page in the url. Then you can enter a valid email, username, and password where a verification email will be sent to you to verify. Once verified, you can login to your account and go to the profile page, or logout. Check the `/review` route to see how auth is handled on that page. You can also attempt to go to the `/profile` route if not logged in, but you will be redirected home. You can try to go to `/login` or `/register` when logged in and you will be redirected home as well.
